docs/infra/api-markdown-documenter/render-api-documentation.mjs

```diff
@@ -78,13 +79,42 @@ export async function renderApiDocumentation(inputDir, outputDir, uriRootDir, ap
 
 	const config = getApiItemTransformationConfigurationWithDefaults({
 		apiModel,
-		documentBoundaries: [
-			ApiItemKind.Class,
-			ApiItemKind.Enum,
-			ApiItemKind.Interface,
-			ApiItemKind.Namespace,
-			ApiItemKind.TypeAlias,
-		],
+		hierarchy: {
```

Comment:
This is the main change. Leverages the new hierarchy config options to move "folder" item documents _into_ their folders, renamed to `index`